### PR TITLE
Support for global variables

### LINKED ISSUE #8: Add support for global variables
This would be really handy to have. A similar interface as `exportFn` could be used, if necessary. I'm fairly new to WASM, but it looks like the exported variables as is are pointers into the memory buffer. So the generated glue code could use type info to build a property accessor or something? If you could provide some guidance on this, I'd be willing to give a go at implementing it.

### PR DESCRIPTION
**Pull request by noxabellus**
Resolves #8

### CODE REVIEW

File: src/generate_js.zig [hunk 1]
```diff
@@ -352,25 +363,100 @@ pub fn main() !void {
...
+                \\
+            ),
+            .f64 => try writer.writeAll(
+                \\        get: () => this.dataView().getFloat64(ptr, true),
+                \\        set: v => this.dataView().setFloat64(ptr, v, true),
+                \\
+            ),
+            .bool => try writer.writeAll(
+                \\        get: () => Boolean(this.dataView().getUint8(ptr, true)),
+                \\        set: v => this.dataView().setUint8(ptr, v, true ? 1 : 0),
```
Comment (scottredig) [inline src/generate_js.zig]: should this be `v  ? 1 : 0, true`?

File: src/zjb.zig [hunk 2]
```diff
@@ -306,6 +313,26 @@ fn validateFromJavascriptArgumentType(comptime T: type) void {
...
+fn validateGlobalType(comptime T: type) void {
+    switch (T) {
+        bool, i32, i64, u32, u64, f32, f64 => {},
+        else => @compileError("unexpected type " ++ @typeName(T) ++ ". Supported types here: bool, i32, i64, u32, u64, f32, f64."),
+    }
+}
+
+fn shortTypeNameGlobal(comptime T: type) []const u8 {
+    return switch (T) {
+        bool => "b",
```
Comment (scottredig) [inline src/zjb.zig]: Going down to one character and using capitalization to indicate length is excessive.  Just use the zig names: `bool`, `i32`, `i64`, etc.

File: example/static/script.js [hunk 3]
```diff
@@ -7,8 +7,20 @@ var zjb = new Zjb();
 
 (function() {
 	WebAssembly.instantiateStreaming(fetch("example.wasm"), {env: env, zjb: zjb.imports}).then(function (results) {
-		zjb.instance = results.instance;
+		zjb.setInstance(results.instance);
```
Comment (scottredig) [inline example/static/script.js:10]: This change needs to be made with the simple example too.  Then build the simple example and update the generated code example in the readme.

**Review by scottredig — CHANGES_REQUESTED** [review]
Great work!  Thanks for working through this feature.  

I've added a few review comments.  Also please add documentation to the README file.

File: src/generate_js.zig (hunk 1 above)
Comment (noxabellus) [reply src/generate_js.zig]: :flushed:

**Pull request merged**